components/automate-ui/src/app/modules/infra-proxy/infra-environment-constraint/infra-environment-constraint.component.html

```diff
@@ -0,0 +1,77 @@
+<div class="constraint-margin">
+  <div class="name-margin">
+    <label>
+      <span class="label">Name</span>
+    </label>
+    <div class="drop">
+      <mat-select panelClass="chef-dropdown" [(value)]="name_id" [disabled]="!constraintKeys.length">
+        <mat-option *ngFor="let list of constraintKeys" [value]="list" [ngClass]="{'checkmark':list==name_id}">
```

Comment:
I think you mean this:
```suggestion
        <mat-option *ngFor="let key of constraintKeys" [value]="key" [ngClass]="{'checkmark': key === name_id}">
```
So: 
1. Change `list` to `key`
2. use `===` not `==` 
3. add whitespace

This applies here and in other occurrences in this file.

Reply:
updated the changes